extensions/netty/runtime/pom.xml

```diff
@@ -22,6 +22,11 @@
             <groupId>io.netty</groupId>
             <artifactId>netty-codec</artifactId>
         </dependency>
+        <dependency>
+            <!-- Required for Brotli -->
+            <groupId>com.aayushatharva.brotli4j</groupId>
+            <artifactId>brotli4j</artifactId>
+        </dependency>
```

Comment:
It's not an optional dependency in Netty?

Reply:
Yes, but needed in the substitutions, so kind of stuck here.

Reply:
I've tried several approaches, even substituting the decoders themselves, it always has references somewhere.

Reply:
If you know how I can delete a field without loading the class that would be awesome! It seems like it always ends up loading it. 

If we delete the class altogether, we need to write substitutions for all the decoders and they are tricky (and there are many). Decoders extend classes which makes substitution particularly annoying.

Reply:
You can condition a substitution to the presence of a class using `onlyWith`. See https://github.com/quarkusio/quarkus/blob/main/extensions/hibernate-reactive/runtime/src/main/java/io/quarkus/hibernate/reactive/runtime/graal/Substitute_FastBootHibernateReactivePersistenceProvider.java#L15 .

Wouldn't it work in this case?

Reply:
To explain a bit better my question: GraalVM is not clever enough to see that the classes are not used if the dependency is not around?

What I'm a bit worried about is that nearly all the Quarkus apps will pay the price of this dependency.

Reply:
It's 2MB+ and what also concerns me is that native compression libraries are keen to CVEs. Now no idea if it will be the case for this one.

Reply:
I've tried the onlyWith, but Netty depends on it, and the native compilation fails because it must load the class. I've tried aliases, does not work either. I've tried to remove the code calling it (decoders) but they extend classes so can't be substituted (and will require looking for all decoders depending on brotli (Found 2 already, probably more)

I understand the price and we need to find a better solution before the final, but I've spent 6 hours on this and failed.

Reply:
(Updating reactive messaging and going back to this)

Reply:
I may have found a way, using an alias on parent class fields, with a different `@TargetClass`, not sure it's going to work, but if it does, that would fix the issue by removing the brotli classes completely.

Reply:
Nope, does not work.

Reply:
Found a way! 

Not totally happy, as I had to substitute code in all decoders, but at least it avoids depending on brotli4j.